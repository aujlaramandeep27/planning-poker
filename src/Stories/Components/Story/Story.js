import React from 'react';

export default function Story({match}) {


    const storyId = match.params.storyId;

    const data = [
        {
            user: 'Anson',
            estimate: 5
        },
        {
            user: 'Nelson',
            estimate: 4
        },
        {
            user: 'Ratul',
            estimate: 3
        },
        {
            user: 'Matt',
            estimate: 2
        },
        {
            user: 'Raman',
            estimate: 1
        },
        {
            user: 'Linna',
            estimate: 3
        }
    ];

    const sum = (a,b) => a + b;

    function average(data) {
        return Math.ceil(data.reduce(sum, 0)/data.length);
    }

    return (
        <div className="uk-container uk-text-center uk-margin-top">
            <h1 className="uk-heading-small uk-heading-divider">Story {storyId}</h1>
            <a href="#/stories/">
                <span uk-icon="icon: arrow-left; ratio: 2" className="uk-position-small uk-position-top-left"></span>
            </a>

            <div className="uk-align-center uk-width-1-2@m">
                <table className="uk-table">
                    <thead>
                        <tr>
                            <th className="uk-text-center">Name</th>
                            <th className="uk-text-center">Estimate</th>
                        </tr>
                    </thead>
                    <tbody>
                        {Array.from(data).map((story, index) =>
                        <tr key={index}>
                            <td>{story.user}</td>
                            <td>{story.estimate}</td>
                        </tr>
                        )}
                        <tr className="uk-text-bold" key={'average'}>
                            <td>Average</td>
                            <td>{average(data.map( item => item.estimate))}</td>
                        </tr>
                    </tbody>
                </table>

                <button className="uk-button uk-button-primary uk-width-3-4">Start/Stop</button>
            </div>
        </div>
    );

}