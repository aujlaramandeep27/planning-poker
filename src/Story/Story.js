import React from 'react';

export default function Story(props) {


    const storyId = props.match.params.storyId;

    const data = [
        {
            user: 'Anson',
            estimate: 5
        },
        {
            user: 'Nelson',
            estimate: 4
        },
        {
            user: 'Ratul',
            estimate: 3
        },
        {
            user: 'Matt',
            estimate: 2
        },
        {
            user: 'Raman',
            estimate: 1
        },
        {
            user: 'Linna',
            estimate: 3
        }
    ];

    return (
        <div className="uk-container uk-text-center uk-margin-top">
            <h1 className="uk-heading-small uk-heading-divider">Story {storyId}</h1>
            <a href="#/host">
                <span uk-icon="icon: arrow-left; ratio: 2" className="uk-position-small uk-position-top-left"></span>
            </a>

            <div className="uk-align-center uk-width-1-2@m">
                <table className="uk-table">
                    <thead>
                        <tr>
                            <th className="uk-text-center">Name</th>
                            <th className="uk-text-center">Estimate</th>
                        </tr>
                    </thead>
                    <tbody>
                        {Array.from(data).map((story, index) =>
                        <tr key={index}>
                            <td>{story.user}</td>
                            <td>{story.estimate}</td>
                        </tr>
                        )}
                        <tr className="uk-text-bold" key={'average'}>
                            <td>Average</td>
                            <td></td>
                        </tr>
                    </tbody>
                </table>
            </div>
        </div>
    );

}