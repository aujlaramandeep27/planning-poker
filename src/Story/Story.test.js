import Story from "./Story";

it('Story renders correctly', () => {
    expect(Story()).toMatchSnapshot();
});