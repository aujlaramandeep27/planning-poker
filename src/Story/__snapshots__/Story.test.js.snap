// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Story renders correctly 1`] = `
<div
  className="uk-container uk-text-center uk-margin-top"
>
  <h1
    className="uk-heading-small uk-heading-divider"
  >
    Story 
    1
  </h1>
  <a
    href="#/host"
  >
    <span
      className="uk-position-small uk-position-top-left"
      uk-icon="icon: arrow-left; ratio: 2"
    />
  </a>
  <div
    className="uk-align-center uk-width-1-2@m"
  >
    <table
      className="uk-table"
    >
      <thead>
        <tr>
          <th
            className="uk-text-center"
          >
            Name
          </th>
          <th
            className="uk-text-center"
          >
            Estimate
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            Anson
          </td>
          <td>
            5
          </td>
        </tr>
        <tr>
          <td>
            Nelson
          </td>
          <td>
            4
          </td>
        </tr>
        <tr>
          <td>
            Ratul
          </td>
          <td>
            3
          </td>
        </tr>
        <tr>
          <td>
            Matt
          </td>
          <td>
            2
          </td>
        </tr>
        <tr>
          <td>
            Raman
          </td>
          <td>
            1
          </td>
        </tr>
        <tr>
          <td>
            Linna
          </td>
          <td>
            3
          </td>
        </tr>
        <tr
          className="uk-text-bold"
        >
          <td>
            Average
          </td>
          <td>
            3
          </td>
        </tr>
      </tbody>
    </table>
    <button
      className="uk-button uk-button-primary uk-width-3-4"
    >
      Start/Stop
    </button>
  </div>
</div>
`;
